code/modules/research/experimentor/experimentor.dm

```diff
@@ -0,0 +1,322 @@
+#define SCANTYPE_POKE 1
+#define SCANTYPE_IRRADIATE 2
+#define SCANTYPE_GAS 3
+#define SCANTYPE_HEAT 4
+#define SCANTYPE_COLD 5
+#define SCANTYPE_OBLITERATE 6
+#define SCANTYPE_DISCOVER 7
+#define SCANTYPE_RECYCLE 8
+
+#define RARITY_COMMON 0
+#define RARITY_UNCOMMON 1
+#define RARITY_RARE 2
+#define RARITY_VERYRARE 3
+
+/////////////////////////EXPERI-MENTOR DEFINITION/////////////////////////////
+//
+//
+//
+
+/obj/machinery/r_n_d/experimentor
+	name = "E.X.P.E.R.I-MENTOR"
+	icon = 'icons/obj/machines/heavy_lathe.dmi'
+	desc = "A marvel of technology, the onboard AI is capable of turning compatible material into brand-new contraptions... sometimes."
+	icon_state = "h_lathe"
+	var/list/allowedobjects = list(/obj/item/discovered_tech, /obj/item/unknown_tech)
+	density = 1
+	anchored = 1
+	use_power = IDLE_POWER_USE
+	var/base_flex_cost = 10
+	var/precise_scanner = 0
+	var/spareparts = 0
+	var/datum/experimentor/loot_definer/LootDefiner = new/datum/experimentor/loot_definer()
+
+/obj/machinery/r_n_d/experimentor/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/experimentor(src)
+	component_parts += new /obj/item/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/stock_parts/manipulator(src)
+	component_parts += new /obj/item/stock_parts/manipulator(src)
+	component_parts += new /obj/item/stock_parts/micro_laser(src)
+	component_parts += new /obj/item/stock_parts/micro_laser(src)
+	RefreshParts()
+
+// Max part rating count is 12 for a max reduction of 4 flex.
+/obj/machinery/r_n_d/experimentor/RefreshParts()
+	var/part_rating_count = -4
+	base_flex_cost = 10
+	for(var/obj/item/stock_parts/manipulator/M in component_parts)
+		part_rating_count += M.rating
+	for(var/obj/item/stock_parts/micro_laser/M in component_parts)
+		part_rating_count += M.rating
+	for(var/obj/item/stock_parts/scanning_module/M in component_parts)
+		precise_scanner = M.rating
+	base_flex_cost -= round(part_rating_count/3,1)
+
+/obj/machinery/r_n_d/experimentor/attackby(obj/item/O, mob/user, params)
+	if(shocked)
+		shock(user,50)
+
+	if(default_deconstruction_screwdriver(user, "h_lathe_maint", "h_lathe", O))
+		return
+
+	if(exchange_parts(user, O))
+		return
+
+	if(panel_open && istype(O, /obj/item/crowbar))
+		default_deconstruction_crowbar(O)
+		return
+
+	if(disabled)
+		return
+
+	if(loaded_item)
+		to_chat(user, "<span class='warning'>The [src] is already loaded.</span>")
+		return
+
+	if(O != null && (istype(O, /obj/item/unknown_tech) || istype(O, /obj/item/discovered_tech)))
```

Comment:
`O != null` can be written as `!O` here

Reply:
istype will also check if it's null by the way. `!O && istype(O, path)` is the same as `istype(O,path)`